### DISCUSSION TITLE
Another use of energy?

### DISCUSSION THREAD

**Discussion by IHOXOHI** [micropython/micropython#16151] [General]
Bueno,
I tested to use a 'super capacitor'  connected on the 3.3... It works. Just a regulator between the vin, a litlle resistor to limitate an execive charged current... In this case, no problem of empty battery... Just connect the usb and it charge the capacitor... For me,  it's perfect.
<my capacitor : 220 F , 4V
Enjoy.
![Uploading DSC_2178.JPG…]()

**Comment by bidrohini** [comment]
Yes. Supercapacitor is helpful in many applications.

**Comment by IHOXOHI** [comment]
Thanks,
But I forgot, NO file system acces during the charge! It could break the board.
It's for that that I have added a DIP... and a VCP mode.

**Comment by sosi-deadeye** [comment]
This is much. If my calculation is not wrong, it should work ~2 hours @4V @100mA. Nice :-)
Fun fact: Capacitor is much bigger as the microcontroller with PCB.

**Comment by IHOXOHI** [comment]
This board consume only 6mA... So, a lots of time...
I love too the pyb.stop function, with the dip interruptor I can keep my board ready during severals days without problems...

**Comment by IHOXOHI** [comment]
Hey,
I tried too with a  'super-capacitor' of 5.5V on a board who had the vin pin, and the 5V pin too (with of course a regulator)... It works fine too...
...
On other hand, I tried a panel solar connected on the 3.3V configuration (VI pin of the regulator), ..., alls on a configuration with a battery and a battery-charger connected... Everything works fine.
I will receive a 3.8V capacitor, in this case it will be possible to mix battery and capacitor... but on the 5V and his board...
I'm questionning about a possible effect of stabilisation on the 3.3V by the capacitor???
Thanks.

**Comment by IHOXOHI** [comment]
![superCapacitor_map](https://github.com/user-attachments/assets/d377909a-e49e-4315-884a-8ad47085983f)

**Reply by IHOXOHI** [reply]
The aim:
you can keep your board plugged.
you can keep your board unplugged.
Just, no file acces during a hard charge statment... I added an ADC (easy on the 3.3!) and a led statement for that... with a litlle resistor between the 2 vins... Perfect?

**Reply by IHOXOHI** [reply]
Other Idea: 2 supercapacitors; one with a litlle capacity but an heavy behaviour, a second with better capacity and light behaviour...? Like this the signal of 3.3V of the board is in holliday!
